Make the notification icon black when its background color is light

snowtimeglass opened the pull request:
## Pull Request template

## Purpose / Description
The notification icon remains white even if the color of the status bar is white or other light one.
![Record_2023-02-11-11-12-21_c428a0327e1e5e78cdd71c6b04816edf_AdobeExpress (1)](https://user-images.githubusercontent.com/10436072/218234530-2c8c69b7-5471-43ee-aac1-3c712a4b89d2.gif)
![image](https://user-images.githubusercontent.com/10436072/218235820-835d0035-ebe4-4b46-a168-efd3db142baa.png)
This commit makes the icon black in such background.

## Fixes
Fixes #13234

## Approach
Make new notification icon files from the current one (`drawable/ic_star_notify.xml`) with Asset Studio (`File` -> `New` -> `Image Asset`)
![image](https://user-images.githubusercontent.com/10436072/218242046-ecacdcab-0363-467f-846d-38928b2ffe0f.png)
![image](https://user-images.githubusercontent.com/10436072/218241137-51c26510-930a-45eb-8e94-0b06261ceb0e.png)
By the process, the original file, `drawable/ic_star_notify.xml`, is deleted automatically. 


## How Has This Been Tested?
Checked on a physical device 
Android Version: 11
Compile SDK Version: 33
![VID_20230211134539_AdobeExpress](https://user-images.githubusercontent.com/10436072/218242079-8e03a009-d944-4489-b3b0-eea6f061415c.gif)
![image](https://user-images.githubusercontent.com/10436072/218244661-4b6bfa6e-9ba7-4d9a-ae41-7835da777a49.png)



## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] ~~You have commented your code, particularly in hard-to-understand areas~~
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] ~~UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)~~

david-allison reviewed (commented):
Do you have documentation that this is the correct way of creating a dark icon?

I agree with the outcome (thank you!!), but I would strongly suspect there's a way to do this using less space via the XML. 

Typically Android pushes us towards XML rather than using .PNGs

snowtimeglass commented:
> Do you have documentation that this is the correct way of creating a dark icon?

No, I don't.
 
> I would strongly suspect there's a way to do this using less space via the XML.


Actually, I have suspected that, too.

At first, I just rewritten the original icon file (`drawable/ic_star_notify.xml`) with the new XML file (and deleted `drawable-anydpi-v24` folder and 4 PNG files, which had newly been added by Asset Studio). The single replacement makes the outcome above properly in my device.

I was about to commit the single change, but I hesitated it because I didn't have the knowledge to justify deleting the `drawable-anydpi-v24` folder and 4 PNG files. In the end, I committed the changes as Asset Studio did.

> Typically Android pushes us towards XML rather than using .PNGs

I agree. I am positive about deleting the PNG files if you think it is appropriate.

david-allison commented:
I'd prefer the single change and will take responsibility if it's the wrong move. Could you rebase it in and I'll get it reviewed

Thanks for the reply!

snowtimeglass commented:
I've rebased and force-pushed it. May these procedures be appropriate.
I've also confirmed that the outcome above is reproduced after this commit.

david-allison commented on AnkiDroid/src/main/res/drawable/ic_star_notify.xml:
```diff
@@ -1,13 +1,18 @@
-      android:fillColor="#ffffff"
-      android:strokeWidth="1"/>
+    android:viewportHeight="124"
+    android:tint="#FFFFFF"
```
I'd have thought that this would have forced `FFFFFF`. Learning experience. Thanks!!

BrayanDSO reviewed (approved):
Thanks!
